File: memento/python/text_memento.py
class TextMemento:
    """
    Saklamak istediğimiz değerlerin tanımlandığı yerdir.
    UML diyagramındaki Memento yapısına denk gelir.
    """
    text: str
    cursor_position: int

    def __init__(self, text: str, cursor_position: int):
        self.text = text
        self.cursor_position = cursor_position

File: memento/go/text-memento.go
package main

// TextMemento Saklamak istediğimiz değerlerin tanımlandığı yerdir.
// UML diyagramındaki Memento yapısına denk gelir.
type TextMemento struct {
	Text           string
	CursorPosition int
}

func NewTextMemento() *TextMemento {
	return &TextMemento{
		Text:           "",
		CursorPosition: 0,
	}
}
